### PR TITLE
Add comp, tesc, and tese as GLSL extensions

### PR DESCRIPTION
**Pull request by multiprecision**
In this pull request, I propose adding [missing GLSL extensions](https://github.com/KhronosGroup/glslang#execution-of-standalone-wrapper):
1. `.comp` compute shader. There are about [13 K files on GitHub](https://github.com/search?utf8=%E2%9C%93&q=extension%3Acomp+NOT+nothack&type=Code).
2. `.tesc` tesselation control shader. There are about [2 K files on GitHub](https://github.com/search?utf8=%E2%9C%93&q=extension%3Atesc+NOT+nothack&type=Code).
3. `.tese` tesselation evaluation shader. There are about [1 K files on GitHub](https://github.com/search?utf8=%E2%9C%93&q=extension%3Atese+NOT+nothack&type=Code).

The samples are from [Sascha Willems' Vulkan C++ examples and demos](https://github.com/SaschaWillems/Vulkan) licensed under MIT License.

### CODE REVIEW

**Comment by pchaigno** [comment]
Neat pull request! Thanks @multiprecision!

I'm not familiar with GLSL, but it looks like there are [other languages than GLSL under the `.comp` extension](https://github.com/search?q=extension%3Acomp+NOT+layout&type=Code&utf8=%E2%9C%93). We'd need to identify them (at least the more common ones) before we add this extension; otherwise, all `.comp` files will be recognized as GLSL.

**Comment by multiprecision** [comment]
It seems they are mostly makefiles and shell scripts.

**Comment by Alhadis** [comment]
@pchaigno I'll get round to fixing the Harvester soon. Honest. 😢

**Comment by Alhadis** [comment]
@multiprecision The `.comp` files that're marked Makefiles and shell-scripts are those with [modelines](https://github.com/chunjie-sam-liu/genome_resequencing_pipeline/blob/ad282db9fc54ee9b41eb00023faaf66dc0bec8c2/software/CNVnator_v0.3.2/root/config/Makefile.comp) and [hashbangs](https://github.com/gvtools/gvtools-legacy/blob/55115bb467891a95a6ae87111dd43a451920f94c/support/grass_sextante_bundle/scripts/r.ecomp.curv/r.ecurv.comp), which are the minority of results (in fact, there were only *four* shell-scripts).

Most of the `.comp` files are unclassified and unidentified. While it's tempting to assume they're all mostly GLSL files, it's possible there're several other unclassified languages out there which *aren't* GLSL (it's happened before).

Basically, we need to be confident that most of these unknown `.comp` files really are GLSL shaders.

**Comment by multiprecision** [comment]
@Alhadis I think most of these unknown `.comp` files are nonstandard? So what? Should I open a new pull request adding just `.tese` and `.tesc`?

**Comment by lildude** [comment]
@multiprecision no need. You can remove the references to `.comp` directly in this PR so it only covers the `.tesc` and `.tese` extensions.

**Comment by larsbrinkhoff** [comment]
Thanks @multiprecision.  You also need to remove the .comp sample file.

**Comment by multiprecision** [comment]
Yep, I'm on it.

**Comment by multiprecision** [comment]
@larsbrinkhoff And done!

**Comment by larsbrinkhoff** [comment]
Looks good.

I would suggest making a separate PR for `.comp`, if you are inclined to research which languaes use this file extension.

**Comment by multiprecision** [comment]
Thanks, @larsbrinkhoff!

I don't know how to prove that most of the 13 K `.comp` files on GitHub are mostly GLSL files.

So far, I have found that `.comp` files are related to:

1. Compiler (compiler metadata, compiler directive, makefile, etc.)
2. Scientific computing ([McStas](http://www.mcstas.org/))
3. Electronic components ([Machinekit](http://www.machinekit.io/docs/hal/comp/))
4. Compressed data (?)

Any idea?

**Comment by larsbrinkhoff** [comment]
No bright ideas, just lots of work.

You can probe the GitHub `.comp` files by doing various searches involving `extension:comp` and interesting keywords.  Then try to exclude some keywords by prefixing `NOT` and see if new languages pop up in the remaining set of files.

You can also download a corpus of `.comp` files.  With some luck, you'll get around 2000 files which you can investigate.  @pchaigno and @Alhadis knows how.

**Comment by Alhadis** [comment]
@multiprecision There [was a script](https://github.com/Alhadis/Harvester) to automate listing and downloading files of a particular extension from public search results. However, it was rushed and hacky; as well as unintuitive. It was broken by recent changes to GitHub's front-end code, so it's currently as useful as a screen-door on a submarine (as we say here).

I intend to redo it properly but can't predict when I'll get around to it. Sorry. 😞

**Comment by larsbrinkhoff** [comment]
Again with the breakage!

**Comment by multiprecision** [comment]
@larsbrinkhoff @Alhadis Searching "extension:comp NOT void" yielded about 9 K files. They are most likely not GLSL compute shader files. Oh well, better not add `.comp` as a file extension then. Thanks!

**Comment by Alhadis** [comment]
[This site](https://fileinfo.com/extension/comp) can often be really useful, BTW. =)

**Comment by lildude** [comment]
I think we're good to merge this now. If you wish to continue pursuing the `.comp` extension side of things, it might be an idea to open a new issue so the discussion can take place there.
